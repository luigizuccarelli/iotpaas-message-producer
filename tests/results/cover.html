
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<title>handlers: Go Coverage Report</title>
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">github.com/luigizuccarelli/iotpaas-message-producer/pkg/handlers/handlers.go (100.0%)</option>
				
				<option value="file1">github.com/luigizuccarelli/iotpaas-message-producer/pkg/validator/validate.go (100.0%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package handlers

import (
        "fmt"
        "io/ioutil"
        "net/http"
        "os"

        "github.com/luigizuccarelli/iotpaas-message-producer/pkg/connectors"
)

const (
        CONTENTTYPE     string = "Content-Type"
        APPLICATIONJSON string = "application/json"
)

// StreamHandler a http response and request for a message producer
func StreamHandler(w http.ResponseWriter, r *http.Request, conn connectors.Clients) <span class="cov8" title="1">{
        var response string

        addHeaders(w, r)
        if r.Method == "OPTIONS" </span><span class="cov8" title="1">{
                w.WriteHeader(http.StatusOK)
                fmt.Fprintf(w, "")
                return
        }</span>

        <span class="cov8" title="1">body, err := ioutil.ReadAll(r.Body)
        if err != nil </span><span class="cov8" title="1">{
                response = `{"statuscode": "500", "status": "ERROR", "message": "Could not read body data"}`
                conn.Error("StreamHandler could not read body data %v", err)
                w.WriteHeader(http.StatusInternalServerError)
        }</span> else<span class="cov8" title="1"> {
                err = conn.SendMessageSync(body)
                if err != nil </span><span class="cov8" title="1">{
                        response = `{"statuscode": "500", "status": "ERROR", "message": "Could not send stream data " + err.Error() +"}`
                        conn.Error("StreamHandler could not send stream data %v", err)
                        w.WriteHeader(http.StatusInternalServerError)
                }</span> else<span class="cov8" title="1"> {
                        response = `{"statuscode": "200", "status": "OK", "message": "Stream data sent successfully"}`
                        w.WriteHeader(http.StatusOK)
                }</span>
        }

        <span class="cov8" title="1">fmt.Fprintf(w, response)</span>
}

// IsAlive a http response and request wrapper for health endpoint checks
// It takes a both response and request objects and returns void
func IsAlive(w http.ResponseWriter, r *http.Request) <span class="cov8" title="1">{
        fmt.Fprintf(w, "{\"version\": \""+os.Getenv("VERSION")+"\"}")
}</span>

// headers (with cors) utility
func addHeaders(w http.ResponseWriter, r *http.Request) <span class="cov8" title="1">{
        w.Header().Set(CONTENTTYPE, APPLICATIONJSON)
        // use this for cors
        w.Header().Set("Access-Control-Allow-Origin", "*")
        w.Header().Set("Access-Control-Allow-Methods", "POST, GET, OPTIONS, PUT, DELETE")
        w.Header().Set("Access-Control-Allow-Headers", "Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization")
}</span>
</pre>
		
		<pre class="file" id="file1" style="display: none">package validator

import (
        "errors"
        "fmt"
        "os"
        "strconv"
        "strings"

        "github.com/microlib/simple"
)

// checkEnvars - private function, iterates through each item and checks the required field
func checkEnvar(item string, logger *simple.Logger) error <span class="cov8" title="1">{
        name := strings.Split(item, ",")[0]
        required, _ := strconv.ParseBool(strings.Split(item, ",")[1])
        logger.Trace(fmt.Sprintf("name %s : required %t", name, required))
        if os.Getenv(name) == "" </span><span class="cov8" title="1">{
                if required </span><span class="cov8" title="1">{
                        logger.Error(fmt.Sprintf("%s envar is mandatory please set it", name))
                        return errors.New(fmt.Sprintf("%s envar is mandatory please set it", name))
                }</span> else<span class="cov8" title="1"> {
                        logger.Error(fmt.Sprintf("%s envar is empty please set it", name))
                }</span>
        }
        <span class="cov8" title="1">return nil</span>
}

// ValidateEnvars : public call that groups all envar validations
// These envars are set via the openshift template
func ValidateEnvars(logger *simple.Logger) error <span class="cov8" title="1">{
        items := []string{
                "LOG_LEVEL,false",
                "SERVER_PORT,true",
                "KAFKA_BROKERS,true",
                "TOPIC,true",
        }
        for x, _ := range items </span><span class="cov8" title="1">{
                if err := checkEnvar(items[x], logger); err != nil </span><span class="cov8" title="1">{
                        return err
                }</span>
        }
        <span class="cov8" title="1">return nil</span>
}
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
