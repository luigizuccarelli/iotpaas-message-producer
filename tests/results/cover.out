mode: set
github.com/luigizuccarelli/iotpaas-message-producer/pkg/validator/validate.go:14.59,18.27 4 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/validator/validate.go:26.2,26.12 1 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/validator/validate.go:18.27,19.15 1 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/validator/validate.go:19.15,22.4 2 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/validator/validate.go:22.9,24.4 1 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/validator/validate.go:31.50,38.26 2 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/validator/validate.go:43.2,43.12 1 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/validator/validate.go:38.26,39.54 1 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/validator/validate.go:39.54,41.4 1 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/handlers/handlers.go:18.85,22.27 3 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/handlers/handlers.go:28.2,29.16 2 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/handlers/handlers.go:45.2,45.26 1 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/handlers/handlers.go:22.27,26.3 3 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/handlers/handlers.go:29.16,33.3 3 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/handlers/handlers.go:33.8,35.17 2 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/handlers/handlers.go:35.17,39.4 3 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/handlers/handlers.go:39.9,42.4 2 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/handlers/handlers.go:50.54,52.2 1 1
github.com/luigizuccarelli/iotpaas-message-producer/pkg/handlers/handlers.go:55.57,61.2 4 1
